### PR TITLE
feat: adding profiles to verified groups added

### PR DESCRIPTION
**Pull request by mfateminia**

### CODE REVIEW

File: libs/groups/src/groups.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@nestjs/common';
+import { Group } from './groups.model';
+import { GroupsMongoService } from './services/groups-mongo.service';
+
+@Injectable()
+export class GroupsService {
+  constructor(private groupsMongo: GroupsMongoService) {}
+
+  public async findBySecret(secret: string) {
```
Comment (Mokto) [inline libs/groups/src/groups.service.ts:9]: findOneBySecret

File: libs/groups/src/groups.controller.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+import { GroupsService } from './groups.service';
+import { Group } from './groups.model';
+
+class CreateGroupDto {
+  @IsArray()
+  @IsString({ each: true })
+  @ApiProperty()
+  public managersUserIds: ObjectID[];
```
Comment (Mokto) [inline libs/groups/src/groups.controller.ts:18]: You can't expect to have ObjectID in DTOs

File: libs/groups/src/groups.controller.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  @ApiPropertyOptional()
+  public url?: string;
+}
+
+@ApiTags('groups')
+@Controller('v1/groups')
+export class GroupsController {
+  constructor(private groups: GroupsService) {}
+
+  @Auth('admin')
```
Comment (Mokto) [inline libs/groups/src/groups.controller.ts:36]: How do you secure this?

Maybe with an API token for now?

File: libs/groups/src/groups.controller.ts (hunk 2 above)
Comment (rasoul-shahriari) [reply libs/groups/src/groups.controller.ts:18]: use @ApiProperty({type: String}) this is the only case you need ApiProperty if it didn't work use @Type()

File: libs/groups/src/groups.controller.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { GroupsService } from './groups.service';
+import { Group } from './groups.model';
+
+class CreateGroupDto {
+  @IsArray()
+  @IsString({ each: true })
+  @ApiProperty()
+  public managersUserIds: ObjectID[];
+  @IsString()
+  @ApiProperty()
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.controller.ts:20]: remove @ApiProperty()

File: libs/groups/src/groups.controller.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  public secret: string;
+  @IsString()
+  @ApiProperty()
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.controller.ts:23]: remove

File: libs/groups/src/groups.controller.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  public name: string;
+  @IsOptional()
+  @IsString()
+  @ApiPropertyOptional()
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.controller.ts:27]: remove

File: libs/groups/src/groups.controller.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import {
+  ApiTags,
+  ApiBody,
+  ApiProperty,
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+import { GroupsService } from './groups.service';
+import { Group } from './groups.model';
+
+class CreateGroupDto {
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.controller.ts:14]: Move all dtos to a model or a .dto file

File: libs/groups/src/groups.controller.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+@ApiTags('groups')
+@Controller('v1/groups')
+export class GroupsController {
+  constructor(private groups: GroupsService) {}
+
+  @Auth('admin')
+  @Post()
+  @ApiBody({ type: CreateGroupDto })
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.controller.ts:38]: I managed to remove a lot of them at the ocean, I guess it's extra, swagger is a big help check every route thers

File: libs/groups/src/groups.model.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { classToClass, Exclude } from 'class-transformer';
+
+export class Group {
+  @ApiProperty({ type: String })
+  public _id: ObjectID;
+  public createdAt?: Date;
+  public updatedAt?: Date;
+  public name: string;
+  @ApiPropertyOptional()
```
Comment (rasoul-shahriari) [inline libs/groups/src/groups.model.ts:11]: remove

File: libs/profiles/src/profiles.controller.ts [hunk 10]
```diff
@@ -7,6 +7,12 @@ import { User } from '@backend/users/users.model';
 import { ProfilesRabbitMQService } from './services/profiles-rabbitmq.service';
 import { ApiTags, ApiBody } from '@nestjs/swagger';
 import { PatchProfilesDto, CreateProfilesDto } from './profiles.dto';
+import { IsString } from 'class-validator';
+
+class AddToGroupDto {
```
Comment (rasoul-shahriari) [inline libs/profiles/src/profiles.controller.ts:12]: move it to profiles.dto

File: libs/profiles/src/profile.model.ts [hunk 11]
```diff
@@ -70,10 +70,12 @@ export class Phone {
 }
 
 export class Profile {
-  @ApiProperty({ type: String })
+  @ApiProperty()
```
Comment (rasoul-shahriari) [inline libs/profiles/src/profile.model.ts:73]: @ApiProperty({ type: String })

File: libs/groups/src/services/groups-mongo.service.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { MongoService } from '@backend/mongo';
+import { ConfigService } from '@backend/config';
+import { Group } from '../groups.model';
+
+const collection = 'groups';
```
Comment (rasoul-shahriari) [inline libs/groups/src/services/groups-mongo.service.ts:6]: And I guess it's really better to move this Inside of your class and use privately for it unless you really want to use it in another place

Comment (mfateminia) [reply libs/groups/src/services/groups-mongo.service.ts:6]: sure, i should do this in a different pr

**Pull request merged**